Show the roster before asking user to login.

lucyhe opened the pull request:
ALMOST READY FOR REVIEW (keeping notes)


Note: these changes are only visible on dev

New:
- clicking "Next" from the splash page goes directly to the roster
- the invite FAB will always be visible
- clicking on the FAB opens a page that looks like the existing login page
- clicking on a network from the new "invite-user" page will prompt the user to login with that social network
- depending on the network, after login, the user will get further instructions for inviting a friend
- "Log out" from the settings panel will log you out of networks, but does not move you back to the splash screen
- if you choose to start copypaste, you will be logged out of all other social networks first
- if the user "hasSeenWelcome" (they've clicked through the welcome messages), they will not see the splash screen again; opening uProxy will always open the roster

Working on these TODOs:
- GitHub invite confirmation does not show until you close the GitHub invite overlay
- The login dialog ("Sign in to x") will show up even if you already logged in to that network
- Quiver login should have some kind of confirmation, instead the invite user overlay just closes
- Update color of "Invitation sent" dialog that we see with GMail invites to be red
- Trying to sign in while you're already signed in seems to log you out

Mocks:
![image](https://cloud.githubusercontent.com/assets/8723127/10800737/e10525a6-7d89-11e5-98d3-4e29f7fd4425.png)

Screenshots:
![image](https://cloud.githubusercontent.com/assets/8723127/10801016/99b74182-7d8b-11e5-8d62-93884e9bc6d7.png)

![image](https://cloud.githubusercontent.com/assets/8723127/10800995/8130f554-7d8b-11e5-9c2e-124e05fc97fc.png)

![image](https://cloud.githubusercontent.com/assets/8723127/10801046/bdda6ee0-7d8b-11e5-9661-a638e86b20b3.png)

![image](https://cloud.githubusercontent.com/assets/8723127/10801049/c1606b82-7d8b-11e5-92c0-76223ce76084.png)

![image](https://cloud.githubusercontent.com/assets/8723127/10800995/8130f554-7d8b-11e5-9c2e-124e05fc97fc.png)

Slight UI weirdness:
![image](https://cloud.githubusercontent.com/assets/8723127/10801061/d18d9af2-7d8b-11e5-8a1f-fcfb15ce59d8.png)

lucyhe commented on src/generic_ui/polymer/contact.html:
```diff
@@ -129,7 +129,7 @@
       <div flex>
         <div class='name' on-tap='{{ toggle }}'><bdi>{{ contact.name }}</bdi></div>
-        <div class='{{ (isOnline || contact.status!=UserStatus.FRIEND) ? "onlineStatus" : "offlineStatus" }}'
+        <div class='{{ isOnline ? "onlineStatus" : "offlineStatus" }}'
```
This was a CSS fix for pending GitHub requests (they were appearing at a lower opacity than expected)

lucyhe commented:
A few more things left to internationalize, but this is ready for review!

jpevarnek commented:
There's a lot of variable initialization taking place within the constructor functions here, any reason for that?
<!-- Reviewable comment -K29PVHMPR9nGTU33gq6:55066041 -->

---

Reviewed 4 of 11 files at r1, 9 of 10 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 31 unresolved discussions, some commit checks failed.

---

<sup>**[src/generic_ui/polymer/contact.html, line 132 \[r2\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K1twz_HphUbm4v8o9r0-r2-132)** ([raw file](https://github.com/uproxy/uproxy/blob/0eb0bc5cb27355457d19db184c0afc66b3abc0eb/src/generic_ui/polymer/contact.html#L132)):</sup>
Was this fixed? (also, it seems to me that it would make sense for pending requests to appear as offline)
<!-- Reviewable comment -K28B6LqIkh5w9em47bb:-1019807100 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 20 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28DiKsZP8VDxFITsYb)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L20)):</sup>
What are you overriding with the `!important`?
<!-- Reviewable comment -K28DiKsZP8VDxFITsYc:-806248246 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 46 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28EHk0FIhK0sR8GSH1)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L46)):</sup>
This (and kind of the size of the file) makes my brain scream "separate element"
<!-- Reviewable comment -K28EHk0FIhK0sR8GSH2:-454985253 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 153 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28DrKdNJs2k3zC986N)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L153)):</sup>
ping on this (and the following similar notes)
<!-- Reviewable comment -K28DrKdNJs2k3zC986O:-1162461982 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 192 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28VYmAt6RzIV9bHF1u)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L192)):</sup>
?
<!-- Reviewable comment -K28VYmBDy19GlhT52H1:166874 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 272 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28VjoH7gSx4F7TeHhn)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L272)):</sup>
It seems like it would be easier to just have a single template that uses `selectedNetworkName` instead of `n`, would their be a problem with that?
<!-- Reviewable comment -K28VjoH7gSx4F7TeHho:758167570 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 10 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29P7TVG1DMpG23FjpH)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L10)):</sup>
I think I saw this function already...
<!-- Reviewable comment -K29KycWEHIbMXs1d8Mr:1647658068 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 16 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Wivpo03tuz1JGu2k)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L16)):</sup>
` :string`
<!-- Reviewable comment -K29Wivpo03tuz1JGu2l:-401314950 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 22 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28WfO0B35Jq1KjWCkj)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L22)):</sup>
Can we do a better type than `any` here?
<!-- Reviewable comment -K28WfO1mweKTsBE6_ND:1122014385 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 29 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28X5QzWAtbhqB2YMwm)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L29)):</sup>
This looks incredibly strange with the indentation, can you either move `core.sendEmail` to the previous line or indent it?
<!-- Reviewable comment -K28X5QzWAtbhqB2YMwn:885399952 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 56 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Wx8Zq2KcUZCe22is)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L56)):</sup>
Can you fill in the todo?
<!-- Reviewable comment -K29Wx8Zq2KcUZCe22it:1841175286 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 123 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28YgMb9FhfGzaE-fLV)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L123)):</sup>
Why the empty line?
<!-- Reviewable comment -K28YgMb9FhfGzaE-fLW:1474625388 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 127 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28YmHiNlkthQJrsXRJ)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L127)):</sup>
I think the norm for `javascript` is to use `for (var i in model.onlineNetworks) {`, any reason you did it this way? (edit: Ben pointed out it may be possible to use for of loops even with our build target now, maybe try that?)
<!-- Reviewable comment -K28YmHiNlkthQJrsXRK:-1257011225 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 143 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28Zpgahf7kV7Wjuagc)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L143)):</sup>
This seems like it would be better to have `initInviteForNetwork` handle logging in if you are not logged in (or just doing the invite if you are).
<!-- Reviewable comment -K28Zpgahf7kV7Wjuagd:384268400 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 180 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28_WYis58DIqGp8Hnu)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L180)):</sup>
I think this is duplicated in `settings.ts`, can you check that?
<!-- Reviewable comment -K28_WYjzRzn3NaFTPET:828324631 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 211 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K299eVcPWiCo1hPmwXI)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L211)):</sup>
This feels like the kind of hack we should be solving by using build flags.  I realize that we do not have a good way to do build flags right now, and am fine with this because of that, but wanted to note it.
<!-- Reviewable comment -K299eVdrItbV8MarUE5:1463908983 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 213 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29JeeJD3UqCQxZ6tcF)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L213)):</sup>
For this and the following functions, can you change the argument name to `networkName`?  That would make it a bit more clear as to what is happening.
<!-- Reviewable comment -K29JeeKSbh-T1Lkabie:-451181117 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 223 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29JpLxzqLX9_RvTTpC)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L223)):</sup>
I'd consider leaning towards `networkNamesChanged` for these function names, just to make things more obvious of why the function is being called (consistent with Polymer's attributeChanged convention)
<!-- Reviewable comment -K29JpLxzqLX9_RvTTpD:511824558 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 230 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KPdCN6doE3BXwG_c)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L230)):</sup>
If this is specifically for Github, can we name it as such?
<!-- Reviewable comment -K29KPdCN6doE3BXwG_d:-1649172838 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 41 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29X_fdH-nAG9eodzps)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L41)):</sup>
Let's not use deep for our own elements, just edit `uproxy-app-bar` so you can supply this.
<!-- Reviewable comment -K29X_fesDEioQA1MRlX:-1375542461 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 48 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Xfu0_SGsFe2dBgP5)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L48)):</sup>
What are you overriding here?
<!-- Reviewable comment -K29Xfu0_SGsFe2dBgP6:1365433340 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 61 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KB3Og_NuUBdm4x1P)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L61)):</sup>
What's our convention for spaces in the `{{` tags?
<!-- Reviewable comment -K29KB3Og_NuUBdm4x1Q:1479296932 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 63 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KFKuzq6OZWavGjeY)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L63)):</sup>
More specific ID?
<!-- Reviewable comment -K29KFKuzq6OZWavGjeZ:-2006564100 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 64 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KGz0EDIHYVg_knsD)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L64)):</sup>
Spaces around comparison operator.
<!-- Reviewable comment -K29KGz0EDIHYVg_knsE:-1909450856 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 80 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KYhJpps6QU-deVwc)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L80)):</sup>
Spaces around comparison operator.
<!-- Reviewable comment -K29KYhJpps6QU-deVwd:-1909450856 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 96 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Kcd4lr0JSbyKegbR)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L96)):</sup>
General comment on the file: It seems like thes two components in here are pretty much ind
<!-- Reviewable comment -K29Kcd4lr0JSbyKegbS:-1581620484 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.ts, line 21 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28WTb503sAWe_IcuCd)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.ts#L21)):</sup>
This looks like the same code as in `invite-user.ts`, is there a way to remove the duplication?
<!-- Reviewable comment -K28WTb6YUuyqQzKk1Ft:-1656186194 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.ts, line 46 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Y3D71_kCxe2SkhWG)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.ts#L46)):</sup>
Let's move this to the previous line?
<!-- Reviewable comment -K29Y3D71_kCxe2SkhWH:-537019963 -->

---

<sup>**[src/generic_ui/scripts/ui.ts, line 210 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Q1IckjyRbGbicDyx)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/scripts/ui.ts#L210)):</sup>
Any reason for the extra line here?
<!-- Reviewable comment -K29Q1IckjyRbGbicDyy:-276769572 -->

---

<sup>**[src/generic_ui/scripts/ui.ts, line 1153 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Q30Gk-_OHe3XkOlu)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/scripts/ui.ts#L1153)):</sup>
We have the same logic in another place, can you move this check somewhere else perchance?
<!-- Reviewable comment -K29Q30HAMd5UntPC_St:481033031 -->

---

<sup>**[src/generic_ui/scripts/ui.ts, line 1233 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29QY89HceBNTdxat7W)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/scripts/ui.ts#L1233)):</sup>
Same comment on loop iteration as above.

Also, it seems like this this will cause us to unnecessarily loop if we do not need to.  How about replacing this function with `this.showInviteControls = this.showRosterBeforeLogin || _.some(this.model.onlineNetworks, (network) => { return this.supportsInvites(network.name); });` (with, you know, some spacing thrown in and my typos corrected)?
<!-- Reviewable comment -K29QY89HceBNTdxat7X:-585260547 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2018)
<!-- Sent from Reviewable.io -->

lucyhe commented:
Review status: 2 of 15 files reviewed at latest revision, 29 unresolved discussions.

---

<sup>**[src/generic_ui/polymer/contact.html, line 132 \[r2\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K1twz_HphUbm4v8o9r0-r2-132)** ([raw file](https://github.com/uproxy/uproxy/blob/0eb0bc5cb27355457d19db184c0afc66b3abc0eb/src/generic_ui/polymer/contact.html#L132)):</sup>
This fixes it. 

I think we should just show what the social provider gives us: if a user is online but not a friend, the social provider should not leak the status to uproxy
<!-- Reviewable comment -K2E1pMeqCPB7IeBIrFD:797167735 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 20 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28DiKsZP8VDxFITsYb)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L20)):</sup>
Oops! Don't think we need that
<!-- Reviewable comment -K2E6Oj6hFpH0HuWsQCp:1159481540 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 46 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28EHk0FIhK0sR8GSH1)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L46)):</sup>
We're going to end up deleting everything that isn't "roster-before-login", so it will only be like this in the transitional phase
<!-- Reviewable comment -K2E6TaVAe_JArewU9Wt:29302472 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 153 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28DrKdNJs2k3zC986N)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L153)):</sup>
Thanks -- I've translated the ones that I think won't change in the near future. Some of the quiver stuff is still untranslated because that will probably change soon!
<!-- Reviewable comment -K2E6cuuWxmqWzJ8yFm3:-1731127370 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 192 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28VYmAt6RzIV9bHF1u)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L192)):</sup>
Removed
<!-- Reviewable comment -K2E6gWPCbQpU_gH012C:610327451 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 272 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28VjoH7gSx4F7TeHhn)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L272)):</sup>
Done.
<!-- Reviewable comment -K2zGZ6nAxUj8y0jjHEm:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 10 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29P7TVG1DMpG23FjpH)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L10)):</sup>
This is also transitional, before the roster-before-login flow is pushed to prod, we will have the old code calling this
<!-- Reviewable comment -K2E7Wub4Pnzs9SEVUYp:-397277227 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 16 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Wivpo03tuz1JGu2k)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L16)):</sup>
Done.
<!-- Reviewable comment -K2E7pwy9BY3XjdJuyLJ:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 22 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28WfO0B35Jq1KjWCkj)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L22)):</sup>
Done.
<!-- Reviewable comment -K2EKdUKOHlyG7fKx3AB:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 29 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28X5QzWAtbhqB2YMwm)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L29)):</sup>
Done.
<!-- Reviewable comment -K2EOeyCP7APSo2bQA4z:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 56 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Wx8Zq2KcUZCe22is)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L56)):</sup>
Done.
<!-- Reviewable comment -K2EOhl29iLmwHcYEWdf:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 123 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28YgMb9FhfGzaE-fLV)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L123)):</sup>
Just to split the functions for the new flow (again because this is transitional). I took it out in case that's what you would prefer.
<!-- Reviewable comment -K2EOl1EsYDpoIdZOb6M:1659513432 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 127 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28YmHiNlkthQJrsXRJ)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L127)):</sup>
Done.
<!-- Reviewable comment -K2EPGM6WRye15DGDVrg:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 143 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28Zpgahf7kV7Wjuagc)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L143)):</sup>
Do you mean change initInviteForNetwork to:

```
    if (!model.getNetwork(networkName)) {
      this.$.loginToInviteFriendDialog.open();
      return;
    }
    this.selectedNetworkName = networkName;
    if (networkName == "GMail" || networkName == "GitHub") {
      // After login for these networks, open another view which allows users
      // to invite their friends.
      this.fire('core-signal', { name: 'open-network-invite-dialog' });
    } else if (networkName == "Facebook-Firebase-V2") {
      this.sendToFacebookFriend();
    }
```

I don't feel that that's too different.
<!-- Reviewable comment -K2EQVx2PUhw6eAxVV9b:1521054896 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 180 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28_WYis58DIqGp8Hnu)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L180)):</sup>
Just checked, doesn't look like it
<!-- Reviewable comment -K2ERLap7e87CvsONUMI:597528509 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 213 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29JeeJD3UqCQxZ6tcF)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L213)):</sup>
Done.
<!-- Reviewable comment -K2ES4g7aZ6yy9e5Oi4Q:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 223 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29JpLxzqLX9_RvTTpC)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L223)):</sup>
Done.
<!-- Reviewable comment -K2EpGfRoe8t4x2yjz3l:-499146009 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 230 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KPdCN6doE3BXwG_c)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L230)):</sup>
Done.
<!-- Reviewable comment -K2EpUjqMXSQqTQl9H-E:-499146009 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 41 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29X_fdH-nAG9eodzps)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L41)):</sup>
Done.
<!-- Reviewable comment -K2EqXcFXf-MJkj1n_M_:-499146009 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 48 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Xfu0_SGsFe2dBgP5)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L48)):</sup>
global.css styles "body /deep/ a"
<!-- Reviewable comment -K2Eq_voFQda8dRLgTUk:-840641162 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 61 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KB3Og_NuUBdm4x1P)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L61)):</sup>
When we move to Polymer 1.0 it has to be no spaces
<!-- Reviewable comment -K2EqcHIZRBDmilP_mJZ:1089612319 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 63 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KFKuzq6OZWavGjeY)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L63)):</sup>
Done.
<!-- Reviewable comment -K2EqsPhrdMRtrp1Jvq0:-499146009 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 64 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KGz0EDIHYVg_knsD)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L64)):</sup>
Done.
<!-- Reviewable comment -K2EqwTu0M3PeOPPKNXF:-499146009 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.html, line 80 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29KYhJpps6QU-deVwc)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.html#L80)):</sup>
Done.
<!-- Reviewable comment -K2EqwzXUNsfLcywmkt5:-499146009 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.ts, line 21 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28WTb503sAWe_IcuCd)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.ts#L21)):</sup>
It will be removed from invite-user.ts once the new flow is launched (because sendToGMailFriend will only be called here)
<!-- Reviewable comment -K2Er7GMNiU034aQ5nKY:1280468811 -->

---

<sup>**[src/generic_ui/polymer/network-invite-user.ts, line 46 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Y3D71_kCxe2SkhWG)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/network-invite-user.ts#L46)):</sup>
Done.
<!-- Reviewable comment -K2ErUnEs5yqOEg5CrLw:-499146009 -->

---

<sup>**[src/generic_ui/scripts/ui.ts, line 210 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Q1IckjyRbGbicDyx)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/scripts/ui.ts#L210)):</sup>
Removed
<!-- Reviewable comment -K2Es6JQswpVUe7g7kMv:610327451 -->

---

<sup>**[src/generic_ui/scripts/ui.ts, line 1153 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29Q30Gk-_OHe3XkOlu)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/scripts/ui.ts#L1153)):</sup>
Since this will be removed soon (TODO above) I'd prefer to keep it simple.  I.e. in the near future this logic will only be needed in networkNamesChanged in invite-user (and not here) so there wont be a duplication problem. 
(I say "simple" because I imagine the alternative is to move code from polymer (networkNamesChanged) into ui.ts, where it wont be required at all, soon)
<!-- Reviewable comment -K2EsWJmEtls_RpbkD7K:1251438178 -->

---

<sup>**[src/generic_ui/scripts/ui.ts, line 1233 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29QY89HceBNTdxat7W)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/scripts/ui.ts#L1233)):</sup>
Done.
<!-- Reviewable comment -K2ytqLlUgIySzUbNCy9:-499146009 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2018)
<!-- Sent from Reviewable.io -->

jpevarnek commented:
Reviewed 13 of 13 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 46 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28EHk0FIhK0sR8GSH1)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L46)):</sup>
Okay, I guess that's fine for now.
<!-- Reviewable comment -K30DfRKqnTNNzVuguF7:1903650997 -->

---

<sup>**[src/generic_ui/polymer/invite-user.html, line 153 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28DrKdNJs2k3zC986N)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.html#L153)):</sup>
Okay, that's fine.  I think it would be good in general if we got into the habit of translating everything immediately, but that's your area.
<!-- Reviewable comment -K30DpmuX-lT7f60r9Kl:-459649807 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 10 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K29P7TVG1DMpG23FjpH)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L10)):</sup>
Okay :(
<!-- Reviewable comment -K30ERVUaA322IdAZGQx:-1891277067 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 123 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28YgMb9FhfGzaE-fLV)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L123)):</sup>
Ehh, would have been fine either way.
<!-- Reviewable comment -K30EfMQlQgbaqhUOAEw:1497480047 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 143 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28Zpgahf7kV7Wjuagc)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L143)):</sup>
Definitely not very different, however, I cannot think of a situation where you would want to call `initInviteForNetwork` and not have it handle the case where the user is not logged in (right now, that does not get checked at all, so I have no idea what would happen if at some later time in the future someone made the mistake of calling it for a network not logged in)
<!-- Reviewable comment -K30F-fj7o_ulHWBiH9C:1828560664 -->

---

<sup>**[src/generic_ui/polymer/invite-user.ts, line 180 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28_WYis58DIqGp8Hnu)** ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L180)):</sup>
My mistake, the code in `settings.ts` (`logOut`), while different, attempts to do the exact same thing though.  Can we refactor them to only be in one place (FWIW, this version is better overall)
<!-- Reviewable comment -K30Fnph9s52TZjb6Jas:-1160129273 -->

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2018)
<!-- Sent from Reviewable.io -->

dborkan commented:
One more small request - when you click "Log-out of uProxy" in the settings menu, can we have the settings menu close?  It used to go back to the splash page, which I don't think we want anymore, but now it's not so obvious what happens because the settings drawer is blocking the roster.  Thanks!
<!-- Reviewable comment -K315s-dINvitR4xmSdK:-1468791135 -->

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2018)
<!-- Sent from Reviewable.io -->

lucyhe commented:
Adding confirmation for logout before copypaste:
![image](https://cloud.githubusercontent.com/assets/8723127/11159004/1c0913ea-8a2a-11e5-9d4a-a6ea9b3ad80a.png)

![image](https://cloud.githubusercontent.com/assets/8723127/11159009/21c73fa0-8a2a-11e5-83d9-99eb79220d9e.png)

lucyhe commented:
Review status: 9 of 17 files reviewed at latest revision, 2 unresolved discussions.

---

_[src/generic_ui/polymer/invite-user.ts, line 143 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28Zpgahf7kV7Wjuagc:-K3Lnlf1HnoeEwc7L1EN:1591658999) ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L143)):_
Ah I tried to change it and I realized it requires fixing a greater race condition in ui.ts. I added a TODO and github issue.

---

_[src/generic_ui/polymer/invite-user.ts, line 180 \[r3\]](https://reviewable.io:443/reviews/uproxy/uproxy/2018#-K28_WYis58DIqGp8Hnu:-K3Lfck-fi-TGin7k2Hf:-572348516) ([raw file](https://github.com/uproxy/uproxy/blob/f34ff0353e77e4be66848441dbc0e329e57828c5/src/generic_ui/polymer/invite-user.ts#L180)):_
Done, moved to ui.ts

---


_Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/uproxy/2018)_
<!-- Sent from Reviewable.io -->

lucyhe commented:
Done, PTAL!
